From 17587ef66d260e6cbcc953feb71874d53141ad67 Mon Sep 17 00:00:00 2001
From: David Wronek <david-w@ctemplar.com>
Date: Wed, 17 Nov 2021 14:59:54 +0100
Subject: [PATCH 202/210] arm64: dts: qcom: msm8916-huawei-y635: Add sound

Headphone button detection may need some tweaking. Vol + is being
recognized as Vol - and Vol - doesn't work at all. Play/Pause works
fine.

Signed-off-by: David Wronek <david-w@ctemplar.com>
---
 .../boot/dts/qcom/msm8916-huawei-y635.dts     | 46 +++++++++++++++++++
 1 file changed, 46 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/msm8916-huawei-y635.dts b/arch/arm64/boot/dts/qcom/msm8916-huawei-y635.dts
index e8ab0fc3ec7..cf0b2fd0fa2 100644
--- a/arch/arm64/boot/dts/qcom/msm8916-huawei-y635.dts
+++ b/arch/arm64/boot/dts/qcom/msm8916-huawei-y635.dts
@@ -4,6 +4,7 @@
 
 #include "msm8916-pm8916.dtsi"
 #include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/sound/apq8016-lpass.h>
 
 / {
 	model = "Huawei Y635";
@@ -129,6 +130,10 @@ &dsi_phy0 {
 	qcom,dsi-phy-regulator-ldo-mode;
 };
 
+&lpass {
+	status = "okay";
+};
+
 &mdss {
 	status = "okay";
 };
@@ -168,6 +173,40 @@ &sdhc_2 {
 	cd-gpios = <&msmgpio 38 GPIO_ACTIVE_LOW>;
 };
 
+&sound {
+	status = "okay";
+
+	pinctrl-names = "default", "sleep";
+	pinctrl-0 = <&cdc_pdm_lines_act>;
+	pinctrl-1 = <&cdc_pdm_lines_sus>;
+
+	model = "msm8916";
+	audio-routing =
+		"AMIC1", "MIC BIAS External1",
+		"AMIC2", "MIC BIAS Internal2",
+		"AMIC3", "MIC BIAS External1";
+
+	dai-link-primary {
+		link-name = "Primary MI2S";
+		cpu {
+			sound-dai = <&lpass MI2S_PRIMARY>;
+		};
+		codec {
+			sound-dai = <&lpass_codec 0>, <&wcd_codec 0>;
+		};
+	};
+
+	dai-link-tertiary {
+		link-name = "Tertiary MI2S";
+		cpu {
+			sound-dai = <&lpass MI2S_TERTIARY>;
+		};
+		codec {
+			sound-dai = <&lpass_codec 1>, <&wcd_codec 1>;
+		};
+	};
+};
+
 &usb {
 	status = "okay";
 	extcon = <&pm8916_usbin>;
@@ -178,6 +217,13 @@ &usb_hs_phy {
 	extcon = <&pm8916_usbin>;
 };
 
+&wcd_codec {
+	qcom,micbias-lvl = <2800>;
+	qcom,mbhc-vthreshold-low = <75 150 237 450 500>;
+	qcom,mbhc-vthreshold-high = <75 150 237 450 500>;
+	qcom,hphl-jack-type-normally-open;
+};
+
 &smd_rpm_regulators {
 	vdd_l1_l2_l3-supply = <&pm8916_s3>;
 	vdd_l4_l5_l6-supply = <&pm8916_s4>;
-- 
2.34.1

